src/coreclr/src/debug/debug-pal/win/diagnosticsipc.cpp

```diff
@@ -193,27 +194,38 @@ IpcStream *IpcStream::DiagnosticsIpc::Connect(ErrorCallback callback)
     return new IpcStream(hPipe, mode);
 }
 
-void IpcStream::DiagnosticsIpc::Close(bool isShutdown, ErrorCallback)
+void IpcStream::DiagnosticsIpc::Close(bool isShutdown, ErrorCallback callback)
 {
     // don't attempt cleanup on shutdown and let the OS handle it
     if (isShutdown)
+    {
+        if (callback != nullptr)
+            callback("Closing without cleaning underlying handles", 100);
```

Comment:
All the ErrorCallbacks just write to the stress log, right? Seeing the codes makes me suspicious that the callback will terminate the pipe and send an error code back, which would cause some bad assumptions below. From looking at the source it seems like it's just a logging thing though.

Reply:
yes, the `ErrorCallback` is just to allow the diag PAL to call back out to `STRESS_LOG`.